### ISSUE TITLE
Split octagon autotuner to separate octagonAnalysis and octagonVars autotuners

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#1449]
Turns out when the octagon autotuner says `Enabled octagon domain for: ...`, it _implicitly_ means "disabling octagon analysis for all other variables, _even if explicitly enabled by the user".
This massively shoot us in the foot for #1394 benchmarks, where explicitly enabling the apron analysis on top of the normal svcomp configuration, which intuitively should only make things more precise, only produced 1 relational invariant. Actually the autotuner more-or-less overrode that, disabling the autotuner gives us 1896 invariants!

We're really selling our novel relational analyses short by disabling them even when explicitly requested. If it's counterintuitive for us, it will definitely be counterintuitive for anyone else evaluating Goblint.

Much less confusing behavior would be to split the autotuner into two:
1. octagonAnalysis, which is just in charge of enabling the octagon analysis, but _not choosing a subset of the variables_.
2. octagonVars, which is in charge of choosing the variables, i.e. _disabling_ octagon analysis for all other variables (because that's what it's really doing, not enabling for additional variables).

**Comment by michael-schwarz** [comment]
Yes, that's a good idea! I was aware that this is the behavior of the autotuner, but then I was also the one advising the thesis which cooked up the autotuners.
If this knowledge has not even propagated in the team, then, as you say, any external people trying it will be confused for sure!
